src/jvm/benchmark/org/jetbrains/kotlinx/lincheck_benchmark/running/LocksBenchmarks.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2023 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck_benchmark.running
+
+import org.jetbrains.kotlinx.lincheck.annotations.Operation
+import org.jetbrains.kotlinx.lincheck_benchmark.AbstractLincheckBenchmark
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.locks.ReentrantLock
+import kotlin.concurrent.withLock
+
+class SpinLockBenchmark : AbstractLincheckBenchmark() {
+    private var counter = 0
+    private val locked = AtomicBoolean()
+
+    @Operation
+    fun increment(): Int = withSpinLock { counter++ }
+
+    @Operation
+    fun get(): Int = withSpinLock { counter }
+
+    private fun<T> withSpinLock(block: () -> T): T {
+        while (!locked.compareAndSet(false, true)) {
+            Thread.yield()
+        }
+        try {
+            return block()
+        } finally {
+            locked.set(false)
+        }
+    }
+}
+
+class ReentrantLockBenchmark : AbstractLincheckBenchmark() {
```

Comment:
I took it from lincheck tests.
There were tests for several kinds of locks: spin locks, intrinsic locks (synchronized), and re-entrant lock objects. 

I just changed the lock-protected data structure from set to counter. Because I think the exact data structure used under lock is not relevant for these benchmarks,  so I picked the simplest one possible. If you have arguments why we should use sets as an underlying data structure, I will be ready to revert it to sets.